library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.cc

```diff
@@ -138,7 +137,16 @@ void PlatformBridgeCertValidator::PendingValidation::verifyCertsByPlatform() {
 void PlatformBridgeCertValidator::PendingValidation::postVerifyResultAndCleanUp(
     bool success, absl::string_view error_details, uint8_t tls_alert,
     OptRef<Stats::Counter> error_counter) {
+  ENVOY_LOG(trace,
+            "Finished platform cert validation for {}, post result callback to network thread",
+            host_name_);
+
+  if (parent_.platform_validator_->release_validator) {
+    parent_.platform_validator_->release_validator();
+  }
   std::weak_ptr<size_t> weak_alive_indicator(parent_.alive_indicator_);
+
+  // Once this task runs, `this` will be deleted so this must be the last statement in the file.
```

Comment:
This seems pretty brittle, is there a way to force the lifetime of `this` or otherwise only capture what's needed here? Is that something you're addressing in https://github.com/envoyproxy/envoy-mobile/pull/2696?

Reply:
Yes, I'm planning to really fix this in #2696 (though that's not complete sorted out yet). The threading/lifetime semantics here I think need to be crisper.